spec/hoteladmin_spec.rb

```diff
@@ -0,0 +1,272 @@
+    it "creates our room objects numbered 1-20" do
+      expect(hotel.build_rooms.length).must_equal 20
+      expect(hotel.build_rooms.first).must_be_instance_of Room
+      expect(hotel.build_rooms).must_be_instance_of Array
+    end
+  end
+
+  describe "HotelAdmin#list_rooms" do
+    it "will return the array room objects" do
+          # 20.times do |index|
+          #   hotel.rooms << Room.new(index+1)
+          # end
+          expect(hotel.list_rooms[0]).must_be_instance_of Room
+          expect(hotel.list_rooms).must_be_instance_of Array
+          expect(hotel.list_rooms.length).must_equal 20
+    end
+  end
+
+  describe "HotelAdmin#retrieve_by_date" do
+    before do
+      hotel.reservations << Reservation.new({guest_id: "SoccerMom2010@gmail.com", room: 1, date_range: (Date.new(2018,10,20)...Date.new(2018,10,22))})
+      hotel.reservations << Reservation.new({guest_id: "Guccifer2.0@ada.com", room: 1, date_range: (Date.new(2018,12,02)...Date.new(2018,12,07))})
+      hotel.reservations << Reservation.new({guest_id: "Jfahmy07@gmail.com", room: 2, date_range: (Date.new(2018,12,03)...Date.new(2018,12,06))})
+    end
+
+    it "raises an argument error if invalid date object is provided" do
+      expect{ (hotel.retrieve_by_date("I am a string")) }.must_raise ArgumentError
+    end
+
+    it "returns an array of the reservations relevant to given date" do
+      reservations_for_date = hotel.retrieve_by_date(Date.new(2018,12,04))
+
+      expect(reservations_for_date).must_be_instance_of Array
+      expect(reservations_for_date.first).must_be_instance_of Reservation
+      expect(reservations_for_date.length).must_equal 2
+    end
+  end
+
+  describe "HotelAdmin#build_reservation" do
+    it "Raises an ArgumentError if items from the Date class are not provided" do
+      expect {
+        hotel.build_reservation("FishandChipsgrl@gmail.com", 4, "2019-01-20", "2019-01-22")
+      }.must_raise ArgumentError
+    end
+  end
+
+  describe "HotelAdmin#reserve_room" do
+    before do
+      @reservation = hotel.build_reservation("FishandChipsgrl@gmail.com", 4, Date.new(2019,01,20), Date.new(2019,01,22))
+    end
+
+    it "returns the new reservation object" do
+      expect(@reservation.guest_id).must_equal "FishandChipsgrl@gmail.com"
+      expect(@reservation.room).must_be_instance_of Integer
+      expect(@reservation.status).must_equal :complete
+      expect(@reservation.date_range).must_equal (Date.new(2019,01,20)...Date.new(2019,01,22))
+      expect(@reservation.stay_cost).must_equal 400.00
+    end
+
+    it "adds new reservation to the hoteladmin object's reservations array" do
+      expect(hotel.reservations.first).must_be_instance_of Reservation
+      expect(hotel.reservations.length).must_equal 1
+    end
+
+    it "will raise an exception if there is an attempt to add a conflicting reservation" do
+      expect{
+        hotel.build_reservation("Mamacita09@gmail.com", 4, Date.new(2019,01,21), Date.new(2019,01,22))
```

Comment:
You should also test what happens when the reservation starts on the date the previous one ended.

Reply:
Fixed.